File: 495. Teemo Attacking.js
/**
 * @param {number[]} timeSeries
 * @param {number} duration
 * @return {number}
 */
var findPoisonedDuration = function(timeSeries, duration) {
    var cnt = duration
    var l = timeSeries.length - 1
    if (l == -1) {
        return 0
    }
    for (var i = 0; i < l; i++) {
        if (timeSeries[i] + duration < timeSeries[i + 1]) {
            cnt += duration
        } else {
            cnt += (timeSeries[i + 1] - timeSeries[i])
        }
    }
    return cnt
};

File: 495. Teemo Attacking.rb
# @param {Integer[]} time_series
# @param {Integer} duration
# @return {Integer}
def find_poisoned_duration(time_series, duration)
    cnt = duration
    l = time_series.length - 1
    i = 0
    if l == -1
        return 0
    end
    while i < l
        if time_series[i] + duration < time_series[i + 1]
            cnt += duration
        else
            cnt += (time_series[i + 1] - time_series[i])
        end
        i += 1
    end
    return cnt
end
